src/ui_fsmenu/keyboard_options.cc

```diff
@@ -25,28 +25,39 @@
 #include "graphic/text_layout.h"
 #include "logic/map_objects/tribes/tribe_descr.h"
 #include "ui_basic/dropdown.h"
+#include "ui_basic/icon.h"
 #include "ui_basic/messagebox.h"
 #include "ui_basic/multilinetextarea.h"
 #include "ui_fsmenu/mousewheel_options.h"
 #include "wlapplication.h"
-#include "wlapplication_options.h"
 
 namespace FsMenu {
 
 constexpr int16_t kPadding = 4;
 constexpr int16_t kButtonHeight = 28;
 
 struct ShortcutChooser : public UI::Window {
-	ShortcutChooser(UI::Panel& parent, const KeyboardShortcut c, Widelands::Game* game_for_fastplace)
-	   : UI::Window(
-	        &parent, UI::WindowStyle::kFsMenu, "choose_shortcut", 0, 0, 300, 200, to_string(c)),
+	ShortcutChooser(KeyboardOptions& parent,
+	                const KeyboardShortcut c,
+	                Widelands::Game* game_for_fastplace)
+	   : UI::Window(&parent.get_topmost_forefather(),
+	                UI::WindowStyle::kFsMenu,
+	                "choose_shortcut",
+	                0,
+	                0,
+	                300,
+	                200,
```

Comment:
Minimum width should be increased:
![fp_chooser](https://user-images.githubusercontent.com/80712255/141217787-9083b776-fc26-4258-8abc-2c48c81698a4.png)

On the other hand, according to the comment in KeyboardOptions::KeyboardOptions, the main keyboard options window's width is set so wide for the old long labels for fastplace, so maybe it could be reduced somewhat, though there are still some long labels for some groups as well as on some other tabs.